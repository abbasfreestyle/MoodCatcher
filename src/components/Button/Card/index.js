import React from 'react';
import PropTypes from 'prop-types';

import { View, Text, StyleSheet } from 'react-native';

import Touchable from 'components/Touchable';

import theme from 'theme';

const styles = StyleSheet.create({
  button: {
    flex: 1,
    padding: 80,
    justifyContent: 'center'
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.secondary,
    borderRadius: 20,
    margin: 20,
    shadowColor: theme.shadow,
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      width: 0,
      height: 2
    },
    elevation: 5
  },
  text: {
    color: theme.onSecondary,
    textAlign: 'center',
    fontSize: 30,
    fontWeight: '400'
  }
});

const Card = props => {
  const { children, onPress } = props;
  return (
    <View style={styles.container}>
      <Touchable onPress={onPress} borderRadius={20}>
        <View style={styles.button}>
          <Text style={styles.text}>{children}</Text>
        </View>
      </Touchable>
    </View>
  );
};

Card.propTypes = {
  children: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
};

export default Card;
