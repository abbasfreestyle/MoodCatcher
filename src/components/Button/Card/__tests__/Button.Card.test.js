import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card from '..';

const requiredProps = {
  children: 'Test Button',
  onPress: jest.fn()
};

const wrapper = shallow(<Card {...requiredProps} />);

describe('Button > Card.', () => {
  describe('Snapshots.', () => {
    test('should match', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
