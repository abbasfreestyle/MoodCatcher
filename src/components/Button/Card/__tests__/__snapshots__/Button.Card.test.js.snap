// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button > Card. Snapshots. should match 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F26AA3",
      "borderRadius": 20,
      "elevation": 5,
      "flex": 1,
      "justifyContent": "center",
      "margin": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 5,
    }
  }
>
  <Touchable
    borderRadius={20}
    onPress={[MockFunction]}
    rippleColor="white"
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "padding": 80,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
            "fontWeight": "400",
            "textAlign": "center",
          }
        }
      >
        Test Button
      </Text>
    </View>
  </Touchable>
</View>
`;
